src/Controller/DatatransResponseController.php

```diff
@@ -95,8 +106,16 @@ public function processSuccessResponse(Request $request, PaymentInterface $payme
    *
    * @throws \Exception
    */
-  public function processErrorResponse(PaymentInterface $payment) {
-
+  public function processErrorResponse(Request $request, PaymentInterface $payment) {
+    if (\Drupal::state()->get('datatrans.debug', FALSE)) {
+      if(\Drupal::moduleHandler()->moduleExists('past')) {
+        past_event_save('datatrans', $payment->id(), $request);
```

Comment:
same. each process method should have a different machine name, so it's easy to identify successful and aborted messages.